src/rez/pip.py

```diff
@@ -267,16 +268,30 @@ def pip_install_package(source_name, pip_version=None, python_version=None,
         tools = []
         src_dst_lut = {}
 
-        for installed_file in distribution.list_installed_files(allow_fail=True):
-            source_file = os.path.normpath(os.path.join(destpath, installed_file[0]))
+        for installed_file in distribution.list_installed_files():
+            # a relative directory that needs to be appended without the
+            # os.sep (via os.path.join) in order for normpath to return the
+            # correct path, this is in order to match the wheel RECORD file
+
+            # Example
+
+            # destpath = /tmp/pip-HBkRGa-rez/re]z_staging/python
+            # installed_file[0] = ../../bin/pydocstyle
+            # normpath of /tmp/pip-HBkRGa-rez/rez_staging/python../../bin/pydocstyle =  /tmp/pip-HBkRGa-rez/rez_staging/bin/pydocstyle OK
+            # normapth of /tmp/pip-HBkRGa-rez/rez_staging/python/../../bin/pydocstyle = /tmp/pip-HBkRGa-rez/bin/pydocstyle WRONG
```

Comment:
The comment and implementation threw me for a little bit here.

The comment should just explain the odd case: that ../../bin/etc is provided by distlib when in fact ../bin/etc seems to be the correct path instead. I was a little confused because the comment here delves into the implementation more than it does explain the situation.

Reply:
Agreed, I was overly verbose aplogies. I will remove the implementation details and reword the comment a bit.